### PR TITLE
enh: add PETPVC wrapper

### PR DESCRIPTION
**Pull request by alexsavio**

### CODE REVIEW

**Comment by chrisfilo** [comment]
A few things:

	* run `make specs` and include the new autogenerated test in this PR
	* add this docstring n top of your file to make the paths right - https://github.com/nipy/nipype/blob/master/nipype/interfaces/c3.py#L3
	* add empty (size == 0) `pet.nii.gz` and `tissues.nii.gz` files to nipype/testing/data/

File: nipype/interfaces/petpvc.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                                use_ext=False, newpath=cwd)
+        return fname
+
+    def _gen_filename(self, name):
+        if name == 'out_file':
+            return self._list_outputs()['out_file']
+        return None
+
+
+if __name__ == '__main__':
```
Comment (chrisfilo) [inline nipype/interfaces/petpvc.py]: no need for a __main___- docstring example will be run by tests

File: nipype/interfaces/petpvc.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+class PETPVC(CommandLine):
+    """ Use PETPVC for partial volume correction of PET images.
+
+    Examples
+    --------
+    >>> from ..testing import example_data
+    >>> #TODO get data for PETPVC
+    >>> pvc = PETPVC()
+    >>> pvc.inputs.in_file   = example_data('pet.nii.gz')
```
Comment (chrisfilo) [inline nipype/interfaces/petpvc.py]: after adding the doctring on top of the file I mentioned in the other comment you can drop `example_data` and just do `'pet.nii.gz'`. Paths will be adjusted so the right file will be picked up.

Comment (chrisfilo) [inline nipype/interfaces/petpvc.py]: after adding the docstring in the beginning of the file as I described in my other comment you can drop the `example_data` and just do `pet.nii.gz`

**Comment by alexsavio** [comment]
Hi,

I get an error when I run `make specs`:
Traceback (most recent call last):
  File "tools/checkspecs.py", line 425, in <module>
    ic.check_modules()
  File "tools/checkspecs.py", line 381, in check_modules
    bad_specs = self.test_specs(m)
  File "tools/checkspecs.py", line 190, in test_specs
    __import__(uri)
  File "/home/alexandre/Software/nipype/nipype/interfaces/dipy/__init__.py", line 3, in <module>
    from .preprocess import Resample, Denoise
  File "/home/alexandre/Software/nipype/nipype/interfaces/dipy/preprocess.py", line 27, in <module>
    from dipy.align.aniso2iso import resample
ImportError: No module named 'dipy.align.aniso2iso'

**Comment by chrisfilo** [comment]
what version of dipy do you use?

**Comment by alexsavio** [comment]
Well, I have it installed from pip, the 0.10.1 version. But it is not in the requirements file either.

**Comment by alexsavio** [comment]
What version should I use?

**Comment by chrisfilo** [comment]
@Garyfallidis is `dipy.align.aniso2iso` something that  was recently added or (re)moved?

**Comment by chrisfilo** [comment]
This errors has been fixed in master. Could you rebase/merge?

**Comment by Garyfallidis** [comment]
Hi @chrisfilo yes we removed it. We had a DeprecationWarning for two releases. I think. Use this instead
`from dipy.align.reslice import reslice` it does the same thing. Let us know if there are any problems.

**Comment by oesteban** [comment]
@chrisfilo I include a final fix on the dipy PR.

**Comment by alexsavio** [comment]
Hi,

I rebased it. Now I get an error when calling `make specs` from nitime. I am using version 0.5, installed with `pip`. I could use another but it is not specified in `requirements.txt`.

To avoid these errors, is there any version specification for all the software packages that nipype wraps?

Here is the error (`File "/home/alexandre/envs/pytre/lib/python3.5/site-packages/nitime/viz.py", line  19, in <module>:  from matplotlib import mpl`).

```
Checking specs and autogenerating spec tests
python tools/checkspecs.py
/home/alexandre/Software/nipype/nipype/__init__.py:66: UserWarning: Running the tests from the install directory may trigger some failures
  warnings.warn('Running the tests from the install directory may '
BEDPOSTX has nonautotest
XFibres has nonautotest
FILMGLS has nonautotest
/home/alexandre/envs/pytre/lib/python3.5/importlib/_bootstrap_external.py:903: FutureWarning: Module nipy.labs.utils.routines deprecated, will be removed
  _imp.create_dynamic, spec)
/home/alexandre/Software/nipype/nipype/interfaces/nipy/model.py:18: FutureWarning: Module nipy.labs.glm deprecated, will be removed. Please use nipy.modalities.fmri.glm instead.
  import nipy.labs.glm.glm as GLM
/home/alexandre/envs/pytre/lib/python3.5/site-packages/matplotlib/__init__.py:1350: UserWarning:  This call to matplotlib.use() has no effect
because the backend has already been chosen;
matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.

  warnings.warn(_use_error_msg)
Traceback (most recent call last):
  File "tools/checkspecs.py", line 425, in <module>
    ic.check_modules()
  File "tools/checkspecs.py", line 381, in check_modules
    bad_specs = self.test_specs(m)
  File "tools/checkspecs.py", line 190, in test_specs
    __import__(uri)
  File "/home/alexandre/Software/nipype/nipype/interfaces/nitime/__init__.py", line 4, in <module>
    from .analysis import (CoherenceAnalyzerInputSpec, CoherenceAnalyzerOutputSpec,
  File "/home/alexandre/Software/nipype/nipype/interfaces/nitime/analysis.py", line 34, in <module>
    import nitime.viz as viz
  File "/home/alexandre/envs/pytre/lib/python3.5/site-packages/nitime/viz.py", line 19, in <module>
    from matplotlib import mpl
ImportError: cannot import name 'mpl'
Makefile:72: recipe for target 'specs' failed
make: *** [specs] Error 1
```

**Comment by oesteban** [comment]
Hi @alexsavio, I suggest you to update to current master in nitime, since they have already fixed that problem (https://github.com/nipy/nitime/blob/master/nitime/viz.py#L19). However, it should give you just a DeprecationWarning, what leads me to think that may be your matplotlib installation is not correct.

**Comment by alexsavio** [comment]
Hi, 

After running `make specs` two other files have been produced that I am not pushing here:
```
#       new file:   nipype/algorithms/tests/test_auto_ErrorMap.py
#       new file:   nipype/algorithms/tests/test_auto_Overlap.py
```

**Pull request merged**